zio-http/src/main/scala/zhttp/service/server/content/compression/CompressionOptions.scala

```diff
@@ -0,0 +1,40 @@
+package zhttp.service.server.content.compression
+
+import io.netty.handler.codec.compression.{CompressionOptions => JCompressionOptions, StandardCompressionOptions}
+import zhttp.service.server.content.compression.CompressionOptions.Kind
+
+final case class CompressionOptions(level: Int, bits: Int, mem: Int, kind: Kind) { self =>
+  def toJava: JCompressionOptions = self.kind match {
+    case CompressionOptions.GZip    => StandardCompressionOptions.gzip(self.level, self.bits, self.mem)
+    case CompressionOptions.Deflate => StandardCompressionOptions.deflate(self.level, self.bits, self.mem)
+  }
+
+}
+
+object CompressionOptions {
+  sealed trait Kind
+  private case object GZip    extends Kind
+  private case object Deflate extends Kind
+
+  /**
+   * Creates GZip CompressionOptions. Defines defaults as per
+   * io.netty.handler.codec.compression.GzipOptions#DEFAULT
+   */
+  def gzip: CompressionOptions = CompressionOptions(6, 15, 8, GZip)
```

Comment:
The reason behind had coding `level, bits, mem` here is the convenience of invoking the method as `CompressionOptions.gzip` over `CompressionOptions.gzip()`

Reply:
Can't we move this as arguments of server settings with these default values